IOMC/RandomEngine/src/RandomNumberGeneratorService.cc

```diff
@@ -1221,6 +1257,16 @@ namespace edm {
         seedL[0] = static_cast<long int>(seed0);
         seedL[1] = static_cast<long int>(seeds[1]);
         labelAndEngine.engine()->setSeeds(seedL,0);
+      } else if ( engineName == "XorShift128Plus" ) {
+        assert(seeds.size() == 4U);        
+        for( unsigned i = 0; i < 4; i++ ) {
+          labelAndEngine.setSeed(seeds[i], i);
+        }
+        long int seedL[4];
+        for( unsigned i = 0; i < 4; ++i ) { 
+          seedL[i] = static_cast<long int>(seeds[i]) ; 
+        }        
+        labelAndEngine.engine()->setSeeds(seedL,0);
```

Comment:
This part is not going to work properly. It is not applying the offsets to the seeds. Probably the best thing is to apply them to the first seed. And it also needs protection against going over the max value and this should be done similar to how TRandom3 does it because the max value is also the maximum possible value for a 32 bit unsigned int. The offsets are important because that is how different streams for multithreading are given different seeds.